### PR TITLE
Add fixes for autovalidate deprecation in Form, FormField, TextFormField, and DropdownButtonFormField

### PR DESCRIPTION
**Pull request by Piinks**
This adds fixes for `autovalidate` in the following classes:
- `Form`
- `FormField`
- `TextFormField`
- `DropdownButtonFormField`

Deprecated in #61648

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/fix_data.yaml [hunk 1]
```diff
@@ -11,6 +11,168 @@
 version: 1
 transforms:
 
+  # Changes made in https://github.com/flutter/flutter/pull/61648
+  - title: 'Migrate to autovalidateMode'
+    date: 2020-09-02
+    element:
+      uris: [ 'widgets.dart' ]
```
Comment (goderbauer) [inline packages/flutter/lib/fix_data.yaml:18]: Isn't this also accessible through material.dart and cupertino.dart?

Comment (goderbauer) [reply packages/flutter/lib/fix_data.yaml:18]: We should do an audit of this file. `material.dart` and `cupertino.dart` export `widget.dart`, so whenever we have something in `widget.dart`, those other two need to be mentioned as well.

@brianegan We talked about this before, I really wish there was an easier way to fill in the uris field.

Comment (goderbauer) [reply packages/flutter/lib/fix_data.yaml:18]: (this also applies to some of the other fixes below and should probably be covered with tests as well)

Comment (Piinks) [reply packages/flutter/lib/fix_data.yaml:18]: I'll update this, and then open another PR that audits the rest and adds tests for those as well.

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by Piinks** [comment]
Manually verified google check is failing due to a separate change. Setting to green. :)

**Pull request merged**